FinSearchUnified/Tests/Helper/StaticHelperTest.php

```diff
@@ -430,5 +468,146 @@ private function updateParentCategoryName(Category $parent, Category $categoryMo
+            'queryString type is corrected' => [
+                null,
+                'originalQuery',
+                'corrected',
+                'corrected',
+                'queryString',
+                'originalQuery'
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider finSmartDidYouMeanProvider
+     *
+     * @param string $didYouMeanQuery
+     * @param string $originalQuery
+     * @param string $queryStringType
+     * @param string $expectedType
+     * @param string $expectedAlternativeQuery
+     * @param string $expectedOriginalQuery
+     */
+    public function testFinSmartDidYouMean(
+        $didYouMeanQuery,
+        $originalQuery,
+        $queryStringType,
+        $expectedType,
+        $expectedAlternativeQuery,
+        $expectedOriginalQuery
+    ) {
+        $finSmartDidYouMean = null;
+
+        // Replicate functionality of StaticHelper::setSmartDidYouMean
+        if (((isset($originalQuery) && $originalQuery !== '') || (isset($didYouMeanQuery) && $didYouMeanQuery !== ''))
+            && ($queryStringType && $queryStringType !== 'forced')) {
+            $type = isset($didYouMeanQuery) ? 'did-you-mean' : $queryStringType;
+            $finSmartDidYouMean = [
+                'type' => $type,
+                'alternative_query' => $type === 'did-you-mean' ? $didYouMeanQuery : 'queryString',
+                'original_query' => $type === 'did-you-mean' ? '' : $originalQuery
+            ];
+        }
+
+        // Create mocked view
+        $view = $this->getMockBuilder(View::class)
+            ->setMethods(['getAssign'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        // Assign the finSmartDidYouMean variable to the mocked view
+        $view->method('getAssign')->willReturnMap([
+            ['finSmartDidYouMean', $finSmartDidYouMean]
+        ]);
+
+        $action = $this->getMockBuilder(Action::class)
+            ->setMethods(['View'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $action->method('View')
+            ->willReturn($view);
+
+        $renderer = $this->getMockBuilder(ViewRenderer::class)
+            ->setMethods(['Action'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $renderer->method('Action')
+            ->willReturn($action);
+
+        $plugin = $this->getMockBuilder(Plugins::class)
+            ->setMethods(['ViewRenderer'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $plugin->method('ViewRenderer')
+            ->willReturn($renderer);
+
+        $front = $this->getMockBuilder(Front::class)
+            ->setMethods(['Plugins'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $front->method('Plugins')
+            ->willReturn($plugin);
+
+        Shopware()->Container()->set('front', $front);
+
+        // Get assigned variable from mocked view object
+        $finSmartDidYouMean = Shopware()
```

Comment:
@ihanli If I am mocking the `View` itself, wont it be necessary to mock the `getAssign` method as well? Because otherwise it will simply return null if we haven't mocked the method. And if we are mocking it then we would need to define a return value as well, which in turn would be testing our mocked data indirectly.

Reply:
@zep-com Not quite. You can test the passed parameter of your mock's `assign` method. So you won't have to mock `getAssign` + `StaticHelper` would use the mocked view object e.g.
```
$view->expects($this->once())
                ->method('assign')
                ->with($this->callback(function ($data) {
                   \PHPUnit_Framework_Assert::assertEquals(
                        'some value',
                        $data['some-key']
                    );
                   
                    return true;
                }));
```